cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NotificationFeedHttpHandler.java

```diff
@@ -63,13 +63,19 @@ public NotificationFeedHttpHandler(Authenticator authenticator, NotificationFeed
   }
 
   @PUT
-  @Path("/feeds/{id}")
-  public void createFeed(HttpRequest request, HttpResponder responder, @PathParam("id") String id) {
+  @Path("/feeds/categories/{feed-category}/names/{feed-name}")
+  public void createFeed(HttpRequest request, HttpResponder responder,
+                         @PathParam("namespace-id") final String namespaceId,
+                         @PathParam("feed-category") final String category,
+                         @PathParam("feed-name") final String name) {
     try {
-      NotificationFeed combinedFeed;
+      Id.NotificationFeed combinedFeed;
       try {
-        NotificationFeed feed = parseBody(request, NotificationFeed.class);
-        combinedFeed = new NotificationFeed.Builder(NotificationFeed.fromId(id))
+        Id.NotificationFeed feed = parseBody(request, Id.NotificationFeed.class);
+        combinedFeed = new Id.NotificationFeed.Builder()
+          .setNamespaceId(namespaceId)
+          .setCategory(category)
+          .setName(name)
           .setDescription(feed == null ? null : feed.getDescription())
```

Comment:
Gson will simply not include the field in the serialized string, when this one is ``null``. Then when deserializing, it will set the field as null, as expected.